collect_app/src/main/java/org/odk/collect/android/dao/FormsDao.java

```diff
@@ -72,6 +72,17 @@ public Cursor getFormsCursor(String[] projection, String selection, String[] sel
     public void deleteFormsDatabase() {
         Collect.getInstance().getContentResolver().delete(FormsProviderAPI.FormsColumns.CONTENT_URI, null, null);
     }
+    
+    public void deleteFormsFromIDs(String[] idsToDelete) {
+
+        String selection = FormsProviderAPI.FormsColumns._ID + " in (";
+        for (int i = 0; i < idsToDelete.length - 1; i++)
+            selection += "?, ";
+        selection += "? )";
+        
+        Collect.getInstance().getContentResolver().delete(FormsProviderAPI.FormsColumns.CONTENT_URI, selection, idsToDelete);
```

Comment:
Oh dear, that would be if you ended up with > 999 forms in your DB that weren't on disk? That would be pretty shocking! How about just leaving a comment saying this would break for > SQLITE_MAX_VARIABLE_NUMBER forms that need deleting? That way we have a record of it but we don't make the code more complex.